src/instigator.rs

```diff
@@ -211,6 +232,59 @@ impl Instigator {
         )
     }
 
+/// Check both pod and its containers' statuses, add failure reasons and messages
+/// to cfg's status field
+    pub fn check_container_statuses_for_pod(
+        &self,
+        pod: Pod,
+        pod_status_map: &mut BTreeMap<String, String>
+    ) {
+        // unbox pod and container options
+        if let Some(pod_status) = pod.status {
```

Comment:
maybe you could use `.and_then()` to reduce these nested `if let`, like this https://github.com/oam-dev/rudr/blob/master/healthscope/src/main.rs#L265

It's just fine if you don't change it.